docgen/parser/parser.go

```diff
@@ -54,6 +58,60 @@ func ParseFile(file *google_protobuf.FileDescriptorProto) (*File, error) {
 	return f, nil
 }
 
+func nestedDescriptorMessages(file google_protobuf.FileDescriptorProto, genFiles map[string]*google_protobuf.FileDescriptorProto) (map[string]*Message, error) {
+	messages := make(map[string]*Message)
+	dependencies := file.GetDependency()
+	pkgName := file.GetPackage()
+	comments := parseComments(file.GetSourceCodeInfo())
+	for _, dep := range dependencies {
+		if _, ok := genFiles[dep]; !ok {
+			// not needed
+			continue
+		}
+		// check if dependency is needed for messages
+		if isDepNeeded(file, genFiles[dep]) {
```

Comment:
invert and early continue

Reply:
Sorry... i'm confused 😅

Reply:
```go
if !isDepNeeded(file, genFiles[dep]) {
     continue
}

dependencyMessages, err := nestedDescriptorMessages(*genFiles[dep], genFiles)
if err != nil {
    return nil, err
}

. . .
```

to decrease the indentation level of the meat of the lgoic

Reply:
oh, ok thanks for this